mud-contracts/frontier-world/src/modules/smart-storage-unit/systems/SmartStorageUnit.sol

```diff
@@ -48,7 +51,7 @@ contract SmartStorageUnit is EveSystem {
    */
   function createAndAnchorSmartStorageUnit(
     uint256 smartObjectId,
-    EntityRecordData memory entityRecordData,
+    EntityRecordTableData memory entityRecordData,
```

Comment:
this should be EntityRecordData

Reply:
Oh I thought we needed to use the autogenerated schema for `EntityRecordData` ; there are other places where I made similar changes

Aren't they the same ?

Reply:
No, 
The autogenerated schemas are only for contract references. To keep the interfaces clean We define the struct  based on what inputs we need from the client and use that struct interface for external communication. 

Reason being 
1. we dont want to convolute the external interface struct with the internal mud table struct data 
eg: In mud.config 
```
 EntityRecordTable: {
      keySchema: {
        entityId: "uint256",
      },
      valueSchema: {
        itemId: "uint256",
        typeId: "uint256",
        volume: "uint256",
        recordExists: "bool",
      },
      tableIdArgument: true,
    },
```
The table definition is for on chain data struct. Here the `recordExists` value is mainly for internal tracking, The clients does not have to worry about passing this data according to the original design. There are many such examples where the table struct has more info which is not necessary for the clients. 

So instead of using the auto generated structs `EntityRecordTableData` its always clean to use a external interface struct with only the inputs we need from clients.

currently in many places we dont follow this but its huge refactoring. so I am just requesting changes only in places where its required. post play test we should make sure everything is consistent

Reply:
fixed in aef1c73